Lots of changes over the past few months

YHordijk opened the pull request:
This PR has grown kind of out of proportions. Here are the other changes I made:
* Fixed an issue with generated vtk files having the wrong file-extension
* Fixed some general import errors
* Fixed `DensfJob` using the old PyOrbb API
* Molecules loaded using the `molecule.load` function now have the xyz-file as part of its properties
* The `DensfJob.output_cub_paths` property now returns a dictionary with the paths
* Fixed an issue with `DensfJob.grid_around_mol` method now uses bohr lengths instead of Angstrom
* Added missing characters in the font-widths data
* We now use the `Orbital.densf_index` from PyOrbb to obtain the correct index
* Fixed an issue with the `resize` CLI program not removing backgrounds from images
* Added the `Result.as_dict` method #497
* Added the first version of the `tcutility.job.WorkFlow` class used for running python scripts as slurm-submittable jobs #498
* Added data about ionic radii of atoms #499
* Added the formatting argument for `log.table` #500
* Fixed reference errors to `cite.cite` decorator #501
* Fixed an issue with citation style flags not being properly used in the `cite` CLI program #501
* Added the `mode` argument to `tcutility.connect.Local.open_file` method #502
* Fixed an issue with the `connect` module trying to ping `Local` server subclasses #503
* Added the `atom_to_index` method for `geometry.MolTransform` class so that we can give atoms as arguments instead of indices #504
* Fixed an issue with paths not being set properly when calculating densities with `DensfJob` #505
* Fixed an issue with the `AMSJob.constraint` leading to incorrect input files #506
* Fixed an issue with `DFTBJob` class not using `plams.Settings` object #507
* Fixed a missing dependency to `jsonpickle`
* Pinning `sphinx` version to be lower than `9.0.0`
* Fixed a typing error

Original description:
We were so far unable to run calculations on Windows machines due to problems with setting the correct environment and a lack of ability to execute the programs we need. This PR fixes this by making use of the `%AMSHOME%\msys\usr\bin\bash.exe` executable that is shipped with AMS. Presumable SCM uses this bash executable to run their calculations on windows as well.

A few steps are required to get AMS running:
1. Set up a `.bat` (batch) file that can be executed by windows. This is similar to bash, but with some small syntactic differences.
2. Set the correct environment in the `.bat` file using the `set` command. The environment should have `AMSHOME` and `AMSBIN` at the minimum.
3. Call the ams program using the `%AMSHOME%\msys\usr\bin\bash.exe` program.
4. Make sure that any paths given in the input files do not contain windows paths. The easiest solution is to copy necessary files over to the calculation directory and using those paths.

Besides that, the input files should remain the same.

SiebeLeDe commented on src/tcutility/job/adf.py:
```diff
@@ -883,7 +888,7 @@ def grid_around_mol(self, mol: plams.Molecule, spacing: float = 0.106, extend: f
             extend: the space between the grid edges and the molecule. Defaults to 7.5 angstrom.
         """
-        coords = mol.as_array()
+        coords = mol.as_array() / 0.529177  # convert coords to bohr
```
Can you use the constants module for this? I think it's better to have all references to constants in once place.

SiebeLeDe commented on src/tcutility/job/generic.py:
Nice, I can confirm that this works on my Windows computer too!

YHordijk commented:
This PR has grown kind of out of proportions. Here are the other changes I made:
* Fixed an issue with generated vtk files having the wrong file-extension
* Fixed some general import errors
* Fixed `DensfJob` using the old PyOrbb API
* Molecules loaded using the `molecule.load` function now have the xyz-file as part of its properties
* The `DensfJob.output_cub_paths` property now returns a dictionary with the paths
* Fixed an issue with `DensfJob.grid_around_mol` method now uses bohr lengths instead of Angstrom
* Added missing characters in the font-widths data
* We now use the `Orbital.densf_index` from PyOrbb to obtain the correct index
* Fixed an issue with the `resize` CLI program not removing backgrounds from images
* Added the `Result.as_dict` method #497
* Added the first version of the `tcutility.job.WorkFlow` class used for running python scripts as slurm-submittable jobs #498
* Added data about ionic radii of atoms #499
* Added the formatting argument for `log.table` #500
* Fixed reference errors to `cite.cite` decorator #501
* Fixed an issue with citation style flags not being properly used in the `cite` CLI program #501
* Added the `mode` argument to `tcutility.connect.Local.open_file` method #502
* Fixed an issue with the `connect` module trying to ping `Local` server subclasses #503
* Added the `atom_to_index` method for `geometry.MolTransform` class so that we can give atoms as arguments instead of indices #504
* Fixed an issue with paths not being set properly when calculating densities with `DensfJob` #505
* Fixed an issue with the `AMSJob.constraint` leading to incorrect input files #506
* Fixed an issue with `DFTBJob` class not using `plams.Settings` object #507
* Fixed a missing dependency to `jsonpickle`
* Pinning `sphinx` version to be lower than `9.0.0`
* Fixed a typing error

SiebeLeDe commented on src/tcutility/job/ams.py:
```diff
@@ -308,7 +308,7 @@ def constraint(self, line: str):
         self._write_constraints()
 
     def _write_constraints(self):
-        s = "\nConstraints\n"
```
Good catch

YHordijk commented on src/tcutility/job/adf.py:
```diff
@@ -962,26 +971,38 @@ def _setup_job(self):
-            runf.write("\n".join(self._preambles) + "\n\n")
-            runf.write(f"sh {self.inputfile_path}\n")
-            runf.write("\n".join(self._postambles))
+        if on_windows:
```
No, the `AMSHOME` and `AMSBIN` variables need to be set in the preambles when running the calculation on Windows
